pkg/discovery/discovery.go

```diff
@@ -357,20 +355,38 @@ func updateStatus(client kubernetes.Interface, namespace string, status string,
 	return setPodStatusAnnotation(client, namespace, podStatus)
 }
 
-func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, pluginResultStatus string, pluginResultCounts map[string]int) error {
+func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, item results.Item) error {
 	podStatus, err := pluginaggregation.GetStatus(client, namespace)
 	if err != nil {
 		return errors.Wrap(err, "failed to get the existing status")
 	}
 
+	integrateResultsIntoStatus(podStatus, pluginType, &item)
+
+	return setPodStatusAnnotation(client, namespace, podStatus)
+}
+
+func integrateResultsIntoStatus(podStatus *pluginaggregation.Status, pluginName string, item *results.Item) {
 	for i := range podStatus.Plugins {
-		if podStatus.Plugins[i].Plugin == pluginType {
-			podStatus.Plugins[i].ResultStatus = pluginResultStatus
-			podStatus.Plugins[i].ResultStatusCounts = pluginResultCounts
-			break
```

Comment:
This was the main idea that was broken before. This was looping over the plugins and updating one value (the first that matched by name) and breaking the loop.

Instead you need to try and update every plugin in the list that matches on name. It is just a slight misnomer since `podStatus.Plugins` is really meaning `podStatus.PluginNodePairs`

This meant I had to add the code (which existed elsewhere) to just snip the `item` at the right spot to get the results for just a particular node.